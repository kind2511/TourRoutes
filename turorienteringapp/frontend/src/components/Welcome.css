.welcome-page {
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    height: 100vh; /* Full viewport height */
    background-color: #f2f2f2;  
    text-align: center;
  }
  
  .welcome-page h1 {
    font-size: 2rem; 
    color: #000000;  
    margin-bottom: 15px; /* Spacing below the head*/
    text-shadow: 2px 2px 3px green; /* Green 3D text shadow */
    font-weight: bold;
  }
  
  .welcome-page p {
    font-size: 1rem; 
    color: #000000; 
    font-weight: bold;
    text-shadow: 2px 2px 3px green; /* Green 3D text shadow */
  }
  