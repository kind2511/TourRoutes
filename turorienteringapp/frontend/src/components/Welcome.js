import React, { useEffect } from 'react';
import { useNavigate } from 'react-router-dom';
import './Welcome.css'; 

function Welcome() {
  const navigate = useNavigate();

  useEffect(() => {
    const timer = setTimeout(() => {
      localStorage.clear(); // Clears all local storage items
      // Then navigate to login page after 5 sec
      navigate('/login');
    }, 5000); 
    return () => {
      clearTimeout(timer); // Cleanup timer
      localStorage.clear(); 
    };
  }, [navigate]);

  return (
    <div className="welcome-page">
      <h1>Welcome to TurRuter</h1>
      <p>Please Wait then login..</p>
    </div>
  );
}

export default Welcome;
