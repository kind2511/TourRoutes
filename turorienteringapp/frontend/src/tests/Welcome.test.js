import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import Welcome from '../components/Welcome';

// Test to check if the Welcome component renders without crashing
test('Welcome component renders without crashing', () => {
    render(
        <BrowserRouter>
            <Welcome />
        </BrowserRouter>
    );
});

//---------------------------------------------->

// Snapshot test for the Welcome component
test('Welcome component snapshot', () => {
    const { asFragment } = render(
      <BrowserRouter>
        <Welcome />
      </BrowserRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
  