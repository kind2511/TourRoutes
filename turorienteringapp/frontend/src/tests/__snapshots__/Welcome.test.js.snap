// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Welcome component snapshot 1`] = `
<DocumentFragment>
  <div
    class="welcome-page"
  >
    <h1>
      Welcome to TurRuter
    </h1>
    <p>
      Please Wait then login..
    </p>
  </div>
</DocumentFragment>
`;
